### PR TITLE
feat(instrumentation): updating dynamodb, fixing bug with metadata flag

### PR DESCRIPTION
**Pull request by ronyis**

### CODE REVIEW

**Review by ranrib — DISMISSED** [review]

**Review by galbash — DISMISSED** [review]
LGTM

File: src/events/aws_sdk.js [hunk 1]
```diff
@@ -322,6 +323,64 @@ const dynamoDBEventCreator = {
...
+            const parametersCopy = { ...parameters };
+            if (config.metadataOnly) {
+                const dataParameters = [
+                    'ScanFilter',
+                    'ExclusiveStartKey',
+                    'ProjectionExpression',
+                    'FilterExpression',
+                    'ExpressionAttributeValues',
+                ];
+                dataParameters.forEach(parameterName => delete parametersCopy[parameterName]);
```
Comment (galbash) [inline src/events/aws_sdk.js]: maybe use parametersCopy.filter? also please make sure this doesn't affect original parameters (shouldn't just test locally or something)

File: src/event.js [hunk 2]
```diff
@@ -44,3 +44,28 @@ module.exports.addToMetadata = function addToMetadata(event, map, fullDataMap =
...
 };
+
+
+/**
+ * Adds JSON serialized object to a resource Metadata
+ * @param {proto.event_pb.Event} event The event to add the items to
+ * @param {string} key The name of field that is added
+ * @param {object} object The object to add
+ * @param {array} [dataFields=[]] List of data fields that should be filtered out
+ *  only if {@link tracer.shouldSendFullData} is True
```
Comment (galbash) [inline src/event.js]: check if this exists

**Review by galbash — APPROVED** [review]

**Pull request merged**